code/game/objects/items/miscellaneous.dm

```diff
@@ -64,6 +63,172 @@
 
 	new /obj/effect/pod_landingzone(get_turf(src), pod)
 
+/obj/item/choice_beacon/ingredient
+	name = "ingredient delivery beacon"
+	desc = "Summon a box of ingredients to help you get started cooking."
+	icon_state = "gangtool-white"
+
+/obj/item/choice_beacon/ingredient/generate_display_names()
+	var/list/ingredients = list()
+	for(var/V in subtypesof(/obj/item/storage/box/ingredients))
+		var/obj/item/storage/box/ingredients/A = V
+		ingredients[initial(A.theme_name)] = A
+	return ingredients
+
+/obj/item/choice_beacon/ingredient/spawn_option(obj/choice,mob/living/M)
+	new choice(get_turf(M))
+	to_chat(M, "<span class='hear'>You hear something crackle from the beacon for a moment before a voice speaks. \"Please stand by for a message from Sophronia Broadcasting. Message as follows: <b>Please enjoy your Sophronia Broadcasting's 'Plasteel Chef' Ingredients Box, exactly as shown in the hit show!</b> Message ends.\"</span>")
+
+/obj/item/storage/box/ingredients //This box is for the randomly chosen version the chef used to spawn with, it shouldn't actually exist.
+	name = "ingredients box"
+	illustration = "fruit"
+	var/theme_name
+
+/obj/item/storage/box/ingredients/Initialize()
+	. = ..()
+	if(theme_name)
+		name = "[name] ([theme_name])"
+		desc = "A box containing supplementary ingredients for the aspiring chef. The box's theme is '[theme_name]'."
+		inhand_icon_state = "syringe_kit"
+
+/obj/item/storage/box/ingredients/wildcard
+	theme_name = "wildcard"
+
+/obj/item/storage/box/ingredients/wildcard/PopulateContents()
+	for(var/i in 1 to 7)
+		var/randomFood = pick(/obj/item/reagent_containers/food/snacks/grown/chili,
+							  /obj/item/reagent_containers/food/snacks/grown/tomato,
+							  /obj/item/reagent_containers/food/snacks/grown/carrot,
+							  /obj/item/reagent_containers/food/snacks/grown/potato,
+							  /obj/item/reagent_containers/food/snacks/grown/potato/sweet,
+							  /obj/item/reagent_containers/food/snacks/grown/apple,
+							  /obj/item/reagent_containers/food/snacks/chocolatebar,
+							  /obj/item/reagent_containers/food/snacks/grown/cherries,
+							  /obj/item/reagent_containers/food/snacks/grown/banana,
+							  /obj/item/reagent_containers/food/snacks/grown/cabbage,
+							  /obj/item/reagent_containers/food/snacks/grown/soybeans,
+							  /obj/item/reagent_containers/food/snacks/grown/corn,
+							  /obj/item/reagent_containers/food/snacks/grown/mushroom/plumphelmet,
+							  /obj/item/reagent_containers/food/snacks/grown/mushroom/chanterelle)
+		new randomFood(src)
```

Comment:
Define randomFood outside the for loop, ya only really need to create it once

Reply:
> Define randomFood outside the for loop, ya only really need to create it once

Oops, good catch. That was something left there by the original creator of the Wildcard Box. I'll fix that up.